bids/ext/reports/cli.py

```diff
@@ -92,24 +102,30 @@ def set_verbosity(verbosity: int | list[int]) -> None:
         LOGGER.setLevel("DEBUG")
 
 
-def cli(argv: Sequence[str] = sys.argv) -> None:
+def cli(args: Sequence[str] = None, namespace=None) -> None:
     """Entry point."""
     parser = base_parser()
+    opts = parser.parse_args(args, namespace)
 
-    args, unknowns = parser.parse_known_args(argv[1:])
-
-    bids_dir = Path(args.bids_dir[0]).resolve()
-    # output_dir = Path(args.output_dir[0])
-    participant_label = args.participant_label or None
+    bids_dir = opts.bids_dir.resolve()
+    output_dir = opts.output_dir.resolve()
+    participant_label = opts.participant_label or None
 
-    set_verbosity(args.verbosity)
+    set_verbosity(opts.verbosity)
 
-    LOGGER.debug(f"{bids_dir}")
+    LOGGER.debug(bids_dir)
 
     layout = BIDSLayout(bids_dir)
 
     report = BIDSReport(layout)
     if participant_label:
-        report.generate(subject=participant_label)
+        counter = report.generate(subject=participant_label)
+    else:
+        counter = report.generate()
+
+    common_patterns = counter.most_common()
+    if not common_patterns:
+        LOGGER.warning("No common patterns found.")
     else:
-        report.generate()
+        with open(output_dir / "report.txt", "w") as f:
+            f.write(str(counter.most_common()[0][0]))
```

Comment:
This will write out the most common report to a text file, but only if there's a description to write out.